### PR TITLE
Change Surface search tolerance.

### PR DESCRIPTION
**Pull request by TomFischer**
Until now the default value of `std::numeric_limits<double>::epsilon()` was used. This PR changes the behaviour, i.e., the user can give some epsilon value different from the default value.

### CODE REVIEW

**Comment by norihiro-w** [comment]
:+1:

File: GeoLib/Surface.h [hunk 1]
```diff
@@ -71,9 +71,11 @@ class Surface final : public GeoObject
     /**
      * is the given point pnt located in the surface
      * @param pnt the point
+     * @param eps geometric tolerance for the search
      * @return true if the point is contained in the surface
      */
-    bool isPntInSfc(MathLib::Point3d const& pnt) const;
+    bool isPntInSfc(MathLib::Point3d const& pnt,
+                    double eps = std::numeric_limits<double>::epsilon()) const;
```
Comment (chleh) [inline GeoLib/Surface.h]: I am against default values. What would be the effort to remove the default?

File: GeoLib/Surface.cpp [hunk 2]
```diff
@@ -153,15 +153,14 @@ bool Surface::isPntInBoundingVolume(MathLib::Point3d const& pnt) const
     return _bounding_volume->containsPoint(pnt);
 }
 
-bool Surface::isPntInSfc(MathLib::Point3d const& pnt) const
+bool Surface::isPntInSfc(MathLib::Point3d const& pnt, double eps) const
```
Comment (chleh) [inline GeoLib/Surface.cpp]: @endJunction would say: "const correctness" :smile:

File: MeshGeoToolsLib/MeshNodeSearcher.cpp [hunk 3]
```diff
@@ -39,10 +39,10 @@ MeshNodeSearcher::MeshNodeSearcher(MeshLib::Mesh const& mesh,
         _mesh(mesh), _mesh_grid(_mesh.getNodes().cbegin(), _mesh.getNodes().cend()),
         _search_length(0.0), _search_all_nodes(search_all_nodes)
 {
-    DBUG("Constructing MeshNodeSearcher obj.");
+    INFO("Constructing MeshNodeSearcher obj.");
```
Comment (chleh) [inline MeshGeoToolsLib/MeshNodeSearcher.cpp]: Is it useful to raise this loglevel? (The raise below probably is)

File: MeshGeoToolsLib/MeshNodesAlongSurface.h [hunk 4]
```diff
@@ -39,7 +39,8 @@ class MeshNodesAlongSurface
      * @param mesh_nodes Nodes the search will be performed on.
      * @param sfc Along the GeoLib::Surface sfc the mesh nodes are searched.
      */
-    MeshNodesAlongSurface(MeshLib::Mesh const& mesh, GeoLib::Surface const& sfc, bool search_all_nodes = true);
+    MeshNodesAlongSurface(MeshLib::Mesh const& mesh, GeoLib::Surface const& sfc,
+                          double epsilon, bool search_all_nodes = true);
```
Comment (chleh) [inline MeshGeoToolsLib/MeshNodesAlongSurface.h]: Looks like the docs need some update. There's another default value here, but that's probably for another PR.

File: GeoLib/Surface.cpp (hunk 2 above)
Comment (TomFischer) [inline GeoLib/Surface.cpp]: Yes, I know. But in my point of view `eps` can not be changed anyway.

File: GeoLib/Surface.h (hunk 1 above)
Comment (TomFischer) [inline GeoLib/Surface.h]: Removed default value.

**Comment by chleh** [comment]
:+1:

**Pull request merged**